sdk/src/packet.rs

```diff
@@ -38,111 +42,143 @@ pub struct Meta {
     pub sender_stake: u64,
 }
 
-#[derive(Clone, Eq)]
-#[repr(C)]
-pub struct Packet {
-    // Bytes past Packet.meta.size are not valid to read from.
-    // Use Packet.data(index) to read from the buffer.
-    buffer: [u8; PACKET_DATA_SIZE],
-    pub meta: Meta,
-}
+pub trait BasePacket: Default + Clone + Sync + Send + Eq + fmt::Debug {
+    const DATA_SIZE: usize;
+    fn populate_packet<T: Serialize>(&mut self, dest: Option<&SocketAddr>, data: &T) -> Result<()>;
 
-impl Packet {
-    pub fn new(buffer: [u8; PACKET_DATA_SIZE], meta: Meta) -> Self {
-        Self { buffer, meta }
-    }
+    fn meta(&self) -> &Meta;
+    fn meta_mut(&mut self) -> &mut Meta;
```

Comment:
What do you think about making the `meta` field private and adding these two methods to `Packet` as an intermediate PR. I realize this is somewhat throwaway work, but would shrink this PR's diff down a little bit too.

Reply:
Yeah sure, whatever helps to isolate the potentially dangerous changes.

Reply:
#29092 for those changes